### ISSUE TITLE
LR Scheduler load_state_dict does not properly update scaling

### ISSUE THREAD

**Issue by tristanbrown** [pytorch/pytorch#52911]
## 🐛 Bug

When using `load_state_dict` to update gamma in a LR scheduler, the LR scaling doesn't actually change, even though the state dict shows the new gamma.

## To Reproduce

```
from torch.optim.lr_scheduler import CyclicLR
scheduler1 = CyclicLR(optimizer, base_lr=0.01, max_lr=0.1, gamma=0.9)

scheduler2 = CyclicLR(optimizer, base_lr=0.01, max_lr=0.1, gamma=1)
scheduler2.load_state_dict(scheduler1.state_dict())

optimizer.step()
scheduler2.step()
print(scheduler2.__dict__)
optimizer.step()
scheduler2.step()
print(scheduler2.__dict__)
```

## Expected behavior
Each step of scheduler2 should show lr values approach a decay of 0.9 with each cycle. However, no decay occurs. Somehow, the gamma=1 value is retained in the scaling function, even though `scheduler2.state_dict()` shows the correct gamma=0.9 value. Switch the gamma values, and the 0.9x decay happens, when there should be no decay, demonstrating that it's a problem with `load_state_dict`, not just a gamma=1 default.

## Environment

 - PyTorch Version (e.g., 1.0): 1.7.1
 - OS (e.g., Linux): Ubuntu 20.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): N/A
 - Python version: 3.8.6
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information:
